vulcanai2/tests/models/test_cnn.py

```diff
@@ -1,114 +1,204 @@
-                        "kernel_size": (5, 5),
-                        "pool_size": 2,
-                        "stride": 2
-                    },
-                    {
-                        "in_channels": 16,
-                        "out_channels": 1,
-                        "kernel_size": (5, 5),
-                        "stride": 2,
-                        "padding": 2
-                    }]
-            },
-            num_classes=3
-        )
-
-    def test_forward_not_nan(self, cnn_noclass):
+    def multi_input_cnn_test_loader(self, multi_input_cnn_data):
+        """Synthetic train data pytorch dataloader object."""
+        test_val = Subset(multi_input_cnn_data,
+                          range(len(multi_input_cnn_data)//2,
+                                len(multi_input_cnn_data)))
+        return DataLoader(test_val, batch_size=2)
+
+    def test_init(self, conv1D_net):
+        """Initialization Test of a ConvNet object."""
+        assert isinstance(conv1D_net, BaseNetwork)
+        assert isinstance(conv1D_net, nn.Module)
+        assert hasattr(conv1D_net, 'network')
+        assert hasattr(conv1D_net, 'in_dim')
+        assert hasattr(conv1D_net, 'record')
+        assert hasattr(conv1D_net, 'device')
+
+        assert conv1D_net._name is not None
+        assert isinstance(conv1D_net._config, ConvNetConfig)
+
+        assert conv1D_net.input_networks is None
+        assert conv1D_net.epoch == 0
+        assert conv1D_net.optim is None
+        assert conv1D_net.criterion is None
+
+        assert not hasattr(conv1D_net, 'metrics')
+
+    def test_function_multi_input(self, conv1D_net, multi_input_cnn):
+        """Test functions wrt multi_input_cnn."""
+        assert isinstance(multi_input_cnn.input_networks, nn.ModuleDict)
+        assert len(list(multi_input_cnn.input_networks)) == 3
+        assert all(multi_input_cnn._get_max_incoming_spatial_dims() == (8, 8, 8))
+        assert multi_input_cnn._merge_input_network_outputs([
+                torch.ones([10, 1, 28, 28]),
+                torch.ones([10, 1, 28, 28, 28]),
+                torch.ones(10, *multi_input_cnn.
+                           input_networks['multi_input_dnn'].out_dim)
+            ]).shape == (10, 3, 8, 8, 8)
+        test_net = pickle.loads(pickle.dumps(multi_input_cnn))
+        test_net._add_input_network(conv1D_net)
+        assert len(list(test_net.input_networks)) == 4
+        assert 'conv1D_net' in test_net.input_networks
+        assert isinstance(test_net.input_networks['conv1D_net'], ConvNet)
+
+    def test_forward(self, conv1D_net):
+        """Test Forward of ConvNet."""
+        out = conv1D_net(torch.ones([10, *conv1D_net.in_dim]))
+        assert out.shape == (10, 64, 1)
+
+    def test_forward_multi_input(self, multi_input_cnn, conv1D_net):
+        """Test Forward of Multi Input ConvNet."""
+        master_device_setter(multi_input_cnn, 'cpu')
+        input_tensor = [
+            torch.ones([10, 1, 28, 28]),
```

Comment:
I'm not completely confident we should be using torch.ones in all of our tests? An alternate might be         np.random.randint(0, 9, size=num_items)

Reply:
the emphasis was on using more than just ones. sure we can use torch.randn but we just need to make sure about making deterministic test cases

Reply:
not sure I got your point... can you point me out what you mean by deterministic test cases please?

Reply:
I think Robert was just pointing out that we need to use "random" numbers, or something better than just ones, but they need to return the same result every time. To do that we need to set a random seed, so there's two things that need to be changed.

Reply:
@rfratila @nirtiac would any of you be interested in trying it out with the deterministic approach? I'm not clear with the idea since I didnt know about it.

Reply:
@priyatharsan she described specifying a random seed and then using randint to randomly create the class targets